File: app/src/main/java/com/theappnazi/notenotifier/receivers/BootReceiver.java
package com.theappnazi.notenotifier.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.theappnazi.notenotifier.services.BootNotifierService;

public class BootReceiver extends BroadcastReceiver {
    public BootReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Intent serviceIntent = new Intent(context, BootNotifierService.class);
            context.startService(serviceIntent);
        }
    }
}

File: app/src/main/java/com/theappnazi/notenotifier/receivers/BootReceiver.kt
package com.theappnazi.notenotifier.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.theappnazi.notenotifier.repository.NoteRepository
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import javax.inject.Inject

@AndroidEntryPoint
class BootReceiver : BroadcastReceiver() {

    @Inject
    lateinit var repository: NoteRepository

    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            val pendingResult = goAsync()
            CoroutineScope(Dispatchers.IO).launch {
                try {
                    repository.restoreNotifications()
                } finally {
                    pendingResult.finish()
                }
            }
        }
    }
}
